src/OAuth2/Storage/Redis.php

```diff
@@ -1,317 +1,77 @@
 <?php
-
 namespace OAuth2\Storage;
 
-use OAuth2\OpenID\Storage\AuthorizationCodeInterface as OpenIDAuthorizationCodeInterface;
-
-/**
- * redis storage for all storage types
- *
- * To use, install "predis/predis" via composer
- *
- * Register client:
- * <code>
- *  $storage = new OAuth2\Storage\Redis($redis);
- *  $storage->setClientDetails($client_id, $client_secret, $redirect_uri);
- * </code>
- */
-class Redis implements AuthorizationCodeInterface,
-    AccessTokenInterface,
-    ClientCredentialsInterface,
-    UserCredentialsInterface,
-    RefreshTokenInterface,
-    JwtBearerInterface,
-    ScopeInterface,
-    OpenIDAuthorizationCodeInterface
+class Redis extends KeyValueAbstract
 {
 
     private $cache;
 
-    /* The redis client */
-    protected $redis;
+    protected $db;
 
-    /* Configuration array */
-    protected $config;
-
-    /**
-     * Redis Storage!
-     *
-     * @param \Predis\Client $redis
-     * @param array          $config
-     */
-    public function __construct($redis, $config=array())
+    public function __construct($connection = array(), array $config = array())
     {
-        $this->redis = $redis;
-        $this->config = array_merge(array(
-            'client_key' => 'oauth_clients:',
-            'access_token_key' => 'oauth_access_tokens:',
-            'refresh_token_key' => 'oauth_refresh_tokens:',
-            'code_key' => 'oauth_authorization_codes:',
-            'user_key' => 'oauth_users:',
-            'jwt_key' => 'oauth_jwt:',
-            'scope_key' => 'oauth_scopes:',
+        if ($connection instanceof \Predis\Client) {
+            $this->db = $connection;
+        } else if (is_array($connection)) {
+            $connection = array_merge(array(
+                'parameters' => null,
+                'options' => null
+            ), $connection);
+            
+            $this->db = new \Predis\Client($connection['parameters'], $connection['options']);
+        } else {
+            throw new \InvalidArgumentException('First argument to ' . __CLASS__ . ' must be an instance of Predis\Client or a configuration array');
+        }
+        
+        $this->config = array_merge($this->config, array(
+            'expire' => 0
         ), $config);
     }
 
-    protected function getValue($key)
+    protected function _makeKey($table, $key)
```

Comment:
Is there a reason for underscore before method name?

Reply:
I removed underscores to conform PSR-2 style